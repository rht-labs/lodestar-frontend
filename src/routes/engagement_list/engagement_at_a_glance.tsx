import React from 'react';
import { Engagement, EngagementStatus } from '../../schemas/engagement_schema';
import { differenceInWeeks, format as formatDate, isValid } from 'date-fns';
import {
  Flex,
  FlexItem,
  Grid,
  GridItem,
  Tooltip,
  TooltipPosition,
} from '@patternfly/react-core';
import {
  ClipboardCheckIcon,
  CodeBranchIcon,
  CubeIcon,
  OutlinedClockIcon,
  UserIcon,
} from '@patternfly/react-icons';
import { ActivityHistoryLineItem } from '../../components/activity_history_line_item/activity_history_line_item';
import { APP_FEATURES } from '../../common/app_features';
import { Feature } from '../../components/feature';
import { EngagementStatusText } from './engagement_status_text';
import { DisplayCreatedByName } from '../../common/display_created_by_name';

function DurationInWeeks({
  startDate,
  endDate,
}: {
  startDate?: Date;
  endDate?: Date;
}) {
  if (!!startDate && isValid(startDate) && !!endDate && isValid(endDate)) {
    return <>{differenceInWeeks(endDate, startDate)}</>;
  } else return <>0</>;
}

export function EngagementAtAGlance({
  engagement,
  status,
}: {
  engagement?: Engagement;
  status?: EngagementStatus;
}) {
  return (
    <>
      <Grid hasGutter>
        <GridItem md={6} lg={12}>
          <Grid hasGutter>
            <GridItem span={12}>
              {'Created by: '}
              <DisplayCreatedByName
                userFromServer={engagement?.creation_details?.created_by_user}
                lastUpdatedBy={engagement?.last_update_by_name}
              />
            </GridItem>
            <GridItem>
              Target start date:{' '}
              {!!engagement?.start_date && isValid(engagement?.start_date)
                ? formatDate(engagement?.start_date, 'MMM dd, yyyy')
                : 'TBA'}
            </GridItem>
          </Grid>
        </GridItem>
        <GridItem md={12} lg={6}>
          <Flex>
            <Flex>
              <FlexItem spacer={{ default: 'spacerSm' }}>
                <Tooltip
                  position={TooltipPosition.bottom}
                  entryDelay={0}
                  content={'Number of people in this engagement'}
                >
                  <UserIcon />
                </Tooltip>
              </FlexItem>
              <FlexItem>{engagement?.engagement_users?.length || 0}</FlexItem>
            </Flex>

            <Flex>
              <FlexItem spacer={{ default: 'spacerSm' }}>
                <Tooltip
                  position={TooltipPosition.bottom}
                  entryDelay={0}
                  content={'Number of weeks for this engagement'}
                >
                  <OutlinedClockIcon />
                </Tooltip>
              </FlexItem>
              <FlexItem>
                <DurationInWeeks
                  startDate={engagement?.start_date}
                  endDate={engagement?.end_date}
                />
              </FlexItem>
            </Flex>
            <Feature name={APP_FEATURES.engagementCardIcons}>
              <>
                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      entryDelay={0}
                      content={'Number of available reports'}
                    >
                      <ClipboardCheckIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>

                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      entryDelay={0}
                      content={'Number of commits'}
                    >
                      <CodeBranchIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>

                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      entryDelay={0}
                      content={'Number of repositories'}
                    >
                      <CubeIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>
              </>
            </Feature>
            <Flex>
              <FlexItem>
                <EngagementStatusText status={status} />
              </FlexItem>
            </Flex>
          </Flex>
        </GridItem>
        <GridItem md={12} lg={6}>
          <ActivityHistoryLineItem commit={engagement?.commits?.[0]} />
        </GridItem>
      </Grid>
    </>
  );
}
