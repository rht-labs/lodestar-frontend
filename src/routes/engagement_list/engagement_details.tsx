import React from 'react';
import { Engagement, EngagementStatus } from '../../schemas/engagement_schema';
import { differenceInWeeks, format as formatDate, isValid } from 'date-fns';
import {
  Flex,
  FlexItem,
  Grid,
  GridItem,
  Tooltip,
  TooltipPosition,
} from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import {
  ClipboardCheckIcon,
  CodeBranchIcon,
  CubeIcon,
  OutlinedClockIcon,
  UserIcon,
} from '@patternfly/react-icons';
import { ActivityHistoryLineItem } from '../../components/activity_history_line_item/activity_history_line_item';
import { APP_FEATURES } from '../../common/app_features';
import { Feature } from '../../components/feature';

function FirstLine({
  status,
  startDate,
  createdBy,
}: {
  status?: EngagementStatus;
  startDate?: Date;
  createdBy?: string;
}) {
  if (status === EngagementStatus.upcoming) {
    return (
      <div data-testid="upcomingEngagement">
        <Grid hasGutter>
          <GridItem span={12}>
            Created by:
            <Link to="#">{!!createdBy ? ' ' + createdBy : 'TBA'}</Link>
          </GridItem>
          <GridItem>
            Target start date:{' '}
            {!!startDate && isValid(startDate)
              ? formatDate(startDate, 'MMM dd, yyyy')
              : 'TBA'}
          </GridItem>
        </Grid>
      </div>
    );
  } else {
    return (
      <div data-testid="activeOrPastEngagement">
        The link to the last demo:
        <Link to="#"> https://pf4.345jdhgksdf.org </Link>
      </div>
    );
  }
}

function DurationInWeeks({
  startDate,
  endDate,
}: {
  startDate?: Date;
  endDate?: Date;
}) {
  if (!!startDate && isValid(startDate) && !!endDate && isValid(endDate)) {
    return <>{differenceInWeeks(endDate, startDate)}</>;
  } else return <>0</>;
}

export function EngagementDetails({
  engagement,
  status,
}: {
  engagement?: Engagement;
  status?: EngagementStatus;
}) {
  return (
    <>
      <Grid hasGutter>
        <GridItem span={12}>
          <FirstLine
            status={status}
            startDate={engagement?.start_date}
            createdBy={engagement?.created_by_user}
          />
        </GridItem>
        <GridItem span={6}>
          <Flex>
            <Flex>
              <FlexItem spacer={{ default: 'spacerSm' }}>
                <Tooltip
                  position={TooltipPosition.bottom}
                  content={'Number of people in this engagement'}
                >
                  <UserIcon />
                </Tooltip>
              </FlexItem>
              <FlexItem>{engagement?.engagement_users?.length || 0}</FlexItem>
            </Flex>

            <Flex>
              <FlexItem spacer={{ default: 'spacerSm' }}>
                <Tooltip
                  position={TooltipPosition.bottom}
                  content={'Number of weeks for this engagement'}
                >
                  <OutlinedClockIcon />
                </Tooltip>
              </FlexItem>
              <FlexItem>
                <DurationInWeeks
                  startDate={engagement?.start_date}
                  endDate={engagement?.end_date}
                />
              </FlexItem>
            </Flex>
            <Feature name={APP_FEATURES.engagementCardIcons}>
              <>
                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      content={'Number of available reports'}
                    >
                      <ClipboardCheckIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>

                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      content={'Number of commits'}
                    >
                      <CodeBranchIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>

                <Flex>
                  <FlexItem spacer={{ default: 'spacerSm' }}>
                    <Tooltip
                      position={TooltipPosition.bottom}
                      content={'Number of repositories'}
                    >
                      <CubeIcon />
                    </Tooltip>
                  </FlexItem>
                  <FlexItem>-</FlexItem>
                </Flex>
              </>
            </Feature>
            <Flex>
              <FlexItem>
                <EngagementStatusText status={status} />
              </FlexItem>
            </Flex>
          </Flex>
        </GridItem>
        <GridItem span={6}>
          <ActivityHistoryLineItem commit={engagement?.commits?.[0]} />
        </GridItem>
      </Grid>
    </>
  );
}

const EngagementStatusText = ({ status }: { status: EngagementStatus }) => {
  const getEngagementStatusText = () => {
    if (status === EngagementStatus.active) {
      return 'Active';
    } else if (status === EngagementStatus.upcoming) {
      return 'Upcoming';
    } else if (status === EngagementStatus.past) {
      return 'Past';
    }
    return '';
  };
  const getStatusColor = (status?: EngagementStatus) => {
    switch (status) {
      case EngagementStatus.upcoming: {
        return '#EC7A08';
      }
      case EngagementStatus.active: {
        return '#6CA100';
      }
      default: {
        return '#B8BBBE';
      }
    }
  };
  return (
    <b style={{ color: getStatusColor(status) }}>
      {getEngagementStatusText().toUpperCase()}
    </b>
  );
};
