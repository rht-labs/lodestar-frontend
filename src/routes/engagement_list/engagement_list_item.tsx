import {
  Engagement,
  getEngagementStatus,
} from '../../schemas/engagement_schema';
import React from 'react';
import { EngagementListItemCard } from './engagement_list_item_card';
import { EngagementAtAGlance } from './engagement_at_a_glance';

export function EngagementListItem({ engagement }: { engagement: Engagement }) {
  const status = getEngagementStatus(engagement);
  return (
    <>
      <EngagementListItemCard
        title={engagement.project_name}
        customer={engagement.customer_name}
        project={engagement.project_name}
      >
        <EngagementAtAGlance engagement={engagement} status={status} />
      </EngagementListItemCard>
    </>
  );
}
