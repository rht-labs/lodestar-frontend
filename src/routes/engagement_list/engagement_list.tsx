import React from 'react';
import { Engagement } from '../../schemas/engagement';
import { EngagementListItem } from './engagement_list_item';
import {
  EmptyState,
  EmptyStateIcon,
  Title,
  EmptyStateBody,
  Button,
} from '@patternfly/react-core';
import { SearchIcon } from '@patternfly/react-icons';
import { useHistory } from 'react-router';

export interface EngagementListProps {
  filter?: (engagement: Engagement) => boolean;
}

function EngagementListEmptyState(props) {
  const history = useHistory();
  return (
    <EmptyState>
      <EmptyStateIcon icon={SearchIcon} />
      <Title headingLevel="h4" size="lg">
        There's nothing here.
      </Title>
      <EmptyStateBody>
        <p style={{ marginBottom: '1rem' }}>
          There are no engagements that match your search criteria.
        </p>
        <Button onClick={() => history.push('/app/engagements/new')}>
          Create New
        </Button>
      </EmptyStateBody>
    </EmptyState>
  );
}

export function EngagementList({ engagements }: { engagements: Engagement[] }) {
  if (!engagements || !engagements.length) {
    return <EngagementListEmptyState />;
  }
  return (
    <>
      {(engagements ?? []).map(e => (
        <EngagementListItem
          key={`${e.project_name}${e.customer_name}`}
          engagement={e}
        />
      ))}
    </>
  );
}
