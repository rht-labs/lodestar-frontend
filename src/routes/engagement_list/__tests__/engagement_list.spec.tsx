import React from 'react';
import { render } from '@testing-library/react';
import { EngagementList } from '../engagement_list';
import { Engagement } from '../../../schemas/engagement';
import { MemoryRouter } from 'react-router';
describe('Engagement list', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <MemoryRouter>
          <EngagementList engagements={[Engagement.fromFake(true)]} />
        </MemoryRouter>
      )
    ).toMatchSnapshot();
    expect(
      render(
        <MemoryRouter>
          <EngagementList engagements={undefined} />
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
