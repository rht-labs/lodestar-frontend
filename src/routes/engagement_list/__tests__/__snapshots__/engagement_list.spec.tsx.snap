// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement list matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        style="margin: 1rem; padding: 2rem 0.5rem;"
      >
        <div
          class="pf-c-card__body"
          style="padding: 0px 2rem;"
        >
          <div
            class="pf-l-grid pf-m-gutter"
            style="align-items: center;"
          >
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-3-col-on-lg"
            >
              <button
                class="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                style="white-space: normal; word-wrap: break-word; text-align: left;"
                type="button"
              >
                <h3
                  class="pf-c-title pf-m-lg"
                  style="font-weight: bolder;"
                >
                  Boots on the Moon
                </h3>
              </button>
              <h4
                class="pf-c-title pf-m-md"
                style="font-weight: normal;"
              >
                NASA
              </h4>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-7-col-on-lg"
            >
              <div
                class="pf-l-grid pf-m-gutter"
              >
                <div
                  class="pf-l-grid__item pf-m-6-col-on-md pf-m-12-col-on-lg"
                >
                  <div
                    class="pf-l-grid pf-m-gutter"
                  >
                    <div
                      class="pf-l-grid__item pf-m-12-col"
                    >
                      Created by: 
                      dwasinge
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      Target start date:
                       
                      Jun 01, 2020
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        1
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        4
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class=""
                      >
                        <b
                          data-cy="upcoming"
                          style="color: rgb(236, 122, 8);"
                        >
                          UPCOMING
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <p>
                    <span>
                      Updated by 
                    </span>
                    <b>
                      random-user@example.com
                       
                    </b>
                    <b>
                      184 days ago
                    </b>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-sm pf-m-4-col-on-md pf-m-2-col-on-lg"
            >
              <button
                class="pf-c-button pf-m-secondary"
                data-cy="view_engagement_button"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                View Engagement
              </button>
            </div>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="pf-c-card pf-m-compact"
      style="margin: 1rem; padding: 2rem 0.5rem;"
    >
      <div
        class="pf-c-card__body"
        style="padding: 0px 2rem;"
      >
        <div
          class="pf-l-grid pf-m-gutter"
          style="align-items: center;"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-md pf-m-3-col-on-lg"
          >
            <button
              class="pf-c-button pf-m-link pf-m-inline"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              style="white-space: normal; word-wrap: break-word; text-align: left;"
              type="button"
            >
              <h3
                class="pf-c-title pf-m-lg"
                style="font-weight: bolder;"
              >
                Boots on the Moon
              </h3>
            </button>
            <h4
              class="pf-c-title pf-m-md"
              style="font-weight: normal;"
            >
              NASA
            </h4>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-md pf-m-7-col-on-lg"
          >
            <div
              class="pf-l-grid pf-m-gutter"
            >
              <div
                class="pf-l-grid__item pf-m-6-col-on-md pf-m-12-col-on-lg"
              >
                <div
                  class="pf-l-grid pf-m-gutter"
                >
                  <div
                    class="pf-l-grid__item pf-m-12-col"
                  >
                    Created by: 
                    dwasinge
                  </div>
                  <div
                    class="pf-l-grid__item"
                  >
                    Target start date:
                     
                    Jun 01, 2020
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
              >
                <div
                  class="pf-l-flex"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-m-spacer-sm"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                          transform=""
                        />
                      </svg>
                    </div>
                    <div
                      class=""
                    >
                      1
                    </div>
                  </div>
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-m-spacer-sm"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                          transform=""
                        />
                      </svg>
                    </div>
                    <div
                      class=""
                    >
                      4
                    </div>
                  </div>
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class=""
                    >
                      <b
                        data-cy="upcoming"
                        style="color: rgb(236, 122, 8);"
                      >
                        UPCOMING
                      </b>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
              >
                <p>
                  <span>
                    Updated by 
                  </span>
                  <b>
                    random-user@example.com
                     
                  </b>
                  <b>
                    184 days ago
                  </b>
                </p>
              </div>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-4-col-on-md pf-m-2-col-on-lg"
          >
            <button
              class="pf-c-button pf-m-secondary"
              data-cy="view_engagement_button"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              View Engagement
            </button>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Engagement list matches snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        style="margin: 1rem; padding: 2rem 0.5rem;"
      >
        <div
          class="pf-c-card__body"
          style="padding: 0px 2rem;"
        >
          <div
            class="pf-l-grid pf-m-gutter"
            style="align-items: center;"
          >
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-3-col-on-lg"
            >
              <button
                class="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                style="white-space: normal; word-wrap: break-word; text-align: left;"
                type="button"
              >
                <h3
                  class="pf-c-title pf-m-lg"
                  style="font-weight: bolder;"
                >
                  Boots on the Moon
                </h3>
              </button>
              <h4
                class="pf-c-title pf-m-md"
                style="font-weight: normal;"
              >
                NASA
              </h4>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-7-col-on-lg"
            >
              <div
                class="pf-l-grid pf-m-gutter"
              >
                <div
                  class="pf-l-grid__item pf-m-6-col-on-md pf-m-12-col-on-lg"
                >
                  <div
                    class="pf-l-grid pf-m-gutter"
                  >
                    <div
                      class="pf-l-grid__item pf-m-12-col"
                    >
                      Created by: 
                      dwasinge
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      Target start date:
                       
                      Jun 01, 2020
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        1
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        4
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class=""
                      >
                        <b
                          data-cy="upcoming"
                          style="color: rgb(236, 122, 8);"
                        >
                          UPCOMING
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <p>
                    <span>
                      Updated by 
                    </span>
                    <b>
                      random-user@example.com
                       
                    </b>
                    <b>
                      184 days ago
                    </b>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-sm pf-m-4-col-on-md pf-m-2-col-on-lg"
            >
              <button
                class="pf-c-button pf-m-secondary"
                data-cy="view_engagement_button"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                View Engagement
              </button>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div>
      <div
        class="pf-c-empty-state"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              transform=""
            />
          </svg>
          <h4
            class="pf-c-title pf-m-lg"
          >
            There's nothing here.
          </h4>
          <div
            class="pf-c-empty-state__body"
          >
            <p
              style="margin-bottom: 1rem;"
            >
              There are no engagements that match your search criteria.
            </p>
            <button
              class="pf-c-button pf-m-primary"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Create New
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-empty-state"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          class="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            transform=""
          />
        </svg>
        <h4
          class="pf-c-title pf-m-lg"
        >
          There's nothing here.
        </h4>
        <div
          class="pf-c-empty-state__body"
        >
          <p
            style="margin-bottom: 1rem;"
          >
            There are no engagements that match your search criteria.
          </p>
          <button
            class="pf-c-button pf-m-primary"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Create New
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
